projects/storefrontlib/shared/components/card/card.component.ts

```diff
@@ -101,6 +129,37 @@ export class CardComponent implements OnInit {
     this.editCard.emit(4);
   }
 
+  save(): void {
+    this.saveMode = false;
+    const saveCardEvent: SaveCardEvent = {
+      saveMode: this.saveMode,
+    };
+
+    Object.keys(this.saveForm.controls).forEach((control) => {
+      if (this.saveForm.get(control)?.touched) {
+        const value = this.saveForm.get(control)?.value;
+        if (value) {
+          saveCardEvent[control] = value;
+        }
+        this.saveForm.get(control)?.markAsUntouched();
+        this.saveForm.get(control)?.markAsPristine();
+      }
+    });
+
+    this.saveCard.emit(saveCardEvent);
+  }
+
+  setFormControlName(name: string | undefined, index: number): string {
+    if (name) {
+      return name.toLocaleLowerCase().replace(/\s/g, '_') + '_' + index;
+    }
+    return '';
```

Comment:
see my previous comment regarding title being optional. another reason to make it mandatory - if you would have several paragraphs with undefined name aka title they would be linked to the same form control, which does not make any sense.